src/app/loadout/loadout-ui/hashtag-utils.ts

```diff
@@ -0,0 +1,21 @@
+import { Loadout } from 'app/loadout/loadout-types';
+
+/**
+ * Compute a mapping of normalized hashtags to loadouts that contain them.
+ * Hashtags are normalized by removing the '#' prefix, replacing underscores with spaces,
+ * and converting to lowercase for case-insensitive grouping.
+ */
+export function computeLoadoutsByHashtag(
+  loadouts: Loadout[],
+  getHashtagsFromLoadout: (loadout: Loadout) => string[],
```

Comment:
You don't need to pass in getHashtagsFromLoadout, you can just depend on the real one.